test/Microsoft.ML.Predictor.Tests/TestTransposer.cs

```diff
@@ -199,9 +199,10 @@ public void TransposerTest()
             using (Transposer trans = Transposer.Create(Env, view, true, 3, 5, 4))
             {
                 // Check to see that A, B, and C were not transposed somehow.
-                Assert.Null(trans.TransposeSchema.GetSlotType(0));
-                Assert.Null(trans.TransposeSchema.GetSlotType(1));
-                Assert.Null(trans.TransposeSchema.GetSlotType(2));
+                var itdv = trans as ITransposeDataView;
```

Comment:
>trans as ITransposeDataView [](start = 27, length = 27)

When we are *certain* that something implements an interface, please just use a `()` style cast. I notice you are not testing it, so, you are not sure.

Look at it this way... If you happen to be wrong with `()`, you will get a cast error. If you happen to be wrong with `as`, you will get a null exception... which could also happen because apparently it is legal for `.TransposeSlotTypes` to be `null` (to judge from other code), and you did not test for that. So it is more direct.

Reply:
Ok. I also checked some other places by finding references to `ITransposeDataView` in VS.

---
In reply to: [248781623](https://github.com/dotnet/machinelearning/pull/2111#discussion_r248781623) [](ancestors = 248781623)